gencode.c

```diff
@@ -661,6 +661,103 @@ syntax(compiler_state_t *cstate)
 }
 
 int
+pcap_is_raw(const char *bpf_string)
+{
+	char sp, separator1 = ',', separator2 = '\n';
+	unsigned short bpf_len;
+
+	if (bpf_string == NULL)
+		return 0;
+
+	if (sscanf(bpf_string, "%hu%c", &bpf_len, &sp) != 2 ||
+			(sp != separator1 && sp != separator2))
+		return 0;
+
+	return 1;
+}
+
+int
+pcap_compile_raw(pcap_t *p, struct bpf_program *program, const char *bpf_string)
+{
+	const char *token;
+	char sp, separator, separator1 = ',', separator2 = '\n';
+	unsigned short bpf_len, i = 0, line = 0;
+	size_t size;
+	struct bpf_insn tmp;
+
+	program->bf_len = 0;
+
+	if (sscanf(bpf_string, "%hu%c", &bpf_len, &sp) != 2 ||
+			(sp != separator1 && sp != separator2) ||
+			bpf_len > BPF_MAXINSNS || bpf_len == 0) {
```

Comment:
That's not a syntax error.

In addition, it requires that `BPF_MAXINSNS` be available in a header file, which it might or might not be, and the header file depends on the OS.  Remember, this has to compile on, at minimum, various versions of:

-  various Linux distributions
- various BSDs, including Darwin
- Solaris
- AIX
- HP-UX

and the header file on the system on which it's compiled might, or might not, match the system on which it's running.

If we're going to impose a limit on the number of instructions (the main reason for which would be to keep from allocating a huge blob of memory and eating up most of the address space or backing store), if we're going to use an `unsigned short`, we might as well just impose a limit of 65535 instructions and let the kernel complain if it has a lower limit.